File: src/main/java/org/alxtek/locaiai/entities/Logement.java
package org.alxtek.locaiai.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;
import org.alxtek.locaiai.enums.TypeLogement;

import java.util.ArrayList;
import java.util.List;

@Entity
@AllArgsConstructor @NoArgsConstructor @Builder
@Getter @Setter
public class Logement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String adresse;
    private String ville;
    private String image;
    private String codePostal;
    @Enumerated(EnumType.STRING)
    private TypeLogement typeLogement;
    @ManyToOne
    @JoinColumn(name = "proprietaire_id")
    @JsonBackReference
    private Utilisateur proprietaire;
    @OneToMany(mappedBy = "logement", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<Annonce> annonces = new ArrayList<>();
}

File: front-end/src/app/interfaces/logement.ts
import {TypeLogement} from "../models/typeLogement.enum";
import {User} from "./utilisateur";
import {Annonce} from "./annonce";

export interface Logement {
  id: number;
  adresse: string;
  ville: string;
  image: string;
  codePostal: string;
  typeLogement: TypeLogement;
  proprietaire: User;
  annonces: Annonce[];
}
